qiskit/transpiler/passes/optimization/optimize_1q_decomposition.py

```diff
@@ -13,15 +13,45 @@
 """Optimize chains of single-qubit gates using Euler 1q decomposer"""
 
 import logging
-from functools import partial
-import numpy as np
+import math
 
 from qiskit.transpiler.basepasses import TransformationPass
 from qiskit.transpiler.passes.utils import control_flow
 from qiskit.quantum_info.synthesis import one_qubit_decompose
+from qiskit._accelerate import euler_one_qubit_decomposer
+from qiskit.circuit.library.standard_gates import (
+    UGate,
+    PhaseGate,
+    U3Gate,
+    U2Gate,
+    U1Gate,
+    RXGate,
+    RYGate,
+    RZGate,
+    RGate,
+    SXGate,
+    XGate,
+)
+from qiskit.circuit import Qubit
+from qiskit.dagcircuit.dagcircuit import DAGCircuit
+
 
 logger = logging.getLogger(__name__)
 
+NAME_MAP = {
+    "u": UGate,
+    "u1": U1Gate,
+    "u2": U2Gate,
+    "u3": U3Gate,
+    "p": PhaseGate,
+    "rx": RXGate,
+    "ry": RYGate,
+    "rz": RZGate,
+    "r": RGate,
+    "sx": SXGate,
+    "x": XGate,
+}
```

Comment:
I think the necessary contents of this mapping are inherently linked to the `VALID_BASIS` const in the Rust file?  If so, can we stick a comment on both to remind us that they need to be kept in sync?

Reply:
It would be nice to have a single source for this information.

Reply:
Also here
https://github.com/Qiskit/qiskit-terra/blob/b6aba599f946a97a9c8d6ee4e76029ecb17749c6/qiskit/quantum_info/synthesis/one_qubit_decompose.py#L43-L56

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/9578/commits/89db1da2d797475094449d93a2e6eb043db58fae